cmd/zdb/zdb.c

```diff
@@ -4574,12 +4581,75 @@ dump_path(char *ds, char *path)
 
 	(void) snprintf(curpath, sizeof (curpath), "dataset=%s path=/", ds);
 
-	err = dump_path_impl(os, root_obj, path);
+	err = dump_path_impl(os, root_obj, path, retobj);
 
 	close_objset(os, FTAG);
 	return (err);
 }
 
+static int
+zdb_copy_object(objset_t *os, uint64_t srcobj, char *destfile)
+{
+	int err = 0;
+	int fd;
+	char *buf = NULL;
+	uint64_t size, readsize, oursize, offset;
+	sa_handle_t *hdl;
+
+	printf("Copying object %" PRIu64 " to file %s\n", srcobj, destfile);
+
+	VERIFY3P(os, ==, sa_os);
+	if ((err = sa_handle_get(os, srcobj, NULL, SA_HDL_PRIVATE, &hdl))) {
+		(void) printf("Failed to get handle for SA znode\n");
+		return (err);
+	}
+	if ((err = sa_lookup(hdl, sa_attr_table[ZPL_SIZE], &size, 8))) {
+		(void) sa_handle_destroy(hdl);
+		return (err);
+	}
+	(void) sa_handle_destroy(hdl);
+
+	printf("Object %" PRIu64 " is %" PRIu64 " bytes\n", srcobj, size);
+	if (size <= 0) {
```

Comment:
Since `size` is unsigned here checking that `== 0` might be clearer than `<= 0`.